src/commands/setup.rs

```diff
@@ -66,9 +66,11 @@ impl Setup {
 
         let (mut hostns, mut netns) =
             core_utils::open_netlink_sockets(&self.network_namespace_path)?;
+        let (mut generic_hostns, mut generic_netns) =
+            core_utils::open_generic_netlink_sockets(&self.network_namespace_path)?;
```

Comment:
I don't think we want this here. As I understand the generic socket is only needed for wireguard and most configs will not use it. We try to be very fast for the common paths.
It also does not make sense to pass it into the driver interface. It would be best to just open the socket in the driver setup call itself.

Reply:
The issue was `&self.network_namespace_path`. 

I could modify the `setup` function on the `NetworkDriver` to pass it in.

Reply:
Or do you have another idea how to pass it down elegantly?

Reply:
Do you even need the host netns socket with wireguard? I think everything is configured in the netns so you would only need this one.

Reply:
Yup, host socket is not needed at all. Simply wanted to keep the function signature the same.
If I implement the Generic for netlink_(generic 🥁) I'd keep it this way and change the invocation to ` let (_, mut generic_netns) =
            core_utils::open_generic_netlink_sockets(&self.network_namespace_path)?;`
Sound good?

Reply:
Updated. Thanks for the hint with using the File Descriptors directly. Worked find and I cleaned up the Driver interface. This in turn also made is easy to get rid of the unwraps the Options introduced